llvm/lib/IR/Instructions.cpp

```diff
@@ -374,10 +374,39 @@ FPClassTest CallBase::getParamNoFPClass(unsigned i) const {
 }
 
 std::optional<ConstantRange> CallBase::getRange() const {
-  const Attribute RangeAttr = getRetAttr(llvm::Attribute::Range);
-  if (RangeAttr.isValid())
-    return RangeAttr.getRange();
-  return std::nullopt;
+  Attribute CBRangeAttr = getRetAttr(llvm::Attribute::Range);
```

Comment:
feels redundant to call the getRetAttr function as that also check the definition if the call do not have the attribute will the following not do what you wanted.
 ```suggestion
  Attribute CBRangeAttr = Attrs.getRetAttr(llvm::Attribute::Range);
```